homeassistant/components/yolink/coordinator.py

```diff
@@ -93,12 +93,14 @@ async def fetch_device_state(self, device: YoLinkDevice):
                     self.data[device.device_id] = device_state_resp.data[
                         ATTR_DEVICE_STATE
                     ]
+                else:
+                    self.data[device.device_id] = {"state": None}
         except YoLinkAuthFailError as yl_auth_err:
             raise ConfigEntryAuthFailed from yl_auth_err
-        except YoLinkClientError as yl_client_err:
-            raise UpdateFailed(
-                f"Error communicating with API: {yl_client_err}"
-            ) from yl_client_err
+        except YoLinkClientError:
+            # Current device unknown state.
+            _LOGGER.error("Error Fetching device: %s state", device.device_id)
```

Comment:
We should raise update failed if there's a general client error. If you want to differentiate between client states you need to raise different errors in the library.

Ie, if missing data is an expected error that we don't want to feature as an unavailable entity, that should have its own exception type in the library and we should not log an error.

Reply:
Since this is an iterative request to get the device status, I don't want the failure of one of the devices to affect the other device status update。